### PR TITLE
Cloud App Library - Phase 1

### PR DESCRIPTION
**Pull request by joeygrover**
With the acceptance of the [Cloud Transport Adapter](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0158-cloud-app-transport-adapter.md) it is necessary to enable cloud and embedded apps to integrate a client side library for SDL. This proposal will best device how to put a library out quickly that is adaptable for both cloud and embedded situations. With that goal in mind, the first phase of cloud/embedded libraries intends to be written in Java. A majority of the code base will be extracted from the SDL Android Library project to start and a common subset of that code will be created to feed into both projects.

### CODE REVIEW

**Comment by joeygrover** [comment]
@jordynmackool ready for review

File: proposals/NNNN-cloud_client_library_phase_1.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+- `SessionType.java` This is supposed to be Service type. 
+- `ByteEnumer.java` This implementation is orphaned and should be removed in favor of standard Enums
+- Any references to PCM or H264/NAV as service types. They should be `AudioStreaming` and `VideoStreaming` 
+- Remove any deprecated code
+
+
+
+### Create `sdl_java_common` library 
+
+Once a working a refactored library has been created, a new library will be extracted from that code base to become the `sdl_java_common` library. This would take into account platform specific APIs that can be used and how each of the classes can be extended in a way that works for all.
```
Comment (jordynmackool) [inline proposals/NNNN-cloud_client_library_phase_1.md]: Change `Once a working a refactored library has been created` to `Once a working refactored library has been created,`

File: proposals/NNNN-cloud_client_library_phase_1.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+After the creation of the common library, the SDL Android library will then need to include the `sdl_java_common` library and make the necessary changes to support it. This is almost guaranteed to be a breaking change so its timing could be pushed from the initial start of the proposal implementation. 
+
+
+## Potential downsides
+
+- Not all cloud applications will be written in Java, likely not even the majority.
+- JVM's will have different performance based on their integrations into specific hardware and operating systems
+- Some code in the `sdl_java_common` library will become a bit fragmented to allow for adaptability between platforms
+- If phased two is agreed upon, there would be two ways to implement SDL into the cloud/embedded. This would require maintenance to both. However, this library would need most of its maintenance for the Android library anyways.
```
Comment (jordynmackool) [inline proposals/NNNN-cloud_client_library_phase_1.md]: Change`phased` to `phase`

Omit the word `anyways`

File: proposals/NNNN-cloud_client_library_phase_1.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
+# Cloud App Library - Phase 1
+
+* Proposal: [SDL-NNNN](NNNN-cloud_client_library_phase_1.md)
+* Author: [Joey Grover](https://github.com/joeygrover)
+* Status: **Awaiting Review**
+* Impacted Platforms: **New: Cloud - Java** 
+
+## Introduction
+
+With the acceptance of the [Cloud Transport Adapter](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0158-cloud-app-transport-adapter.md) it is necessary to enable cloud and embedded apps to integrate a client side library for SDL. This proposal will best device how to put a library out quickly that is adaptable for both cloud and embedded situations. With that goal in mind, the first phase of cloud/embedded libraries intends to be written in Java. A majority of the code base will be extracted from the SDL Android Library project to start and a common subset of that code will be created to feed into both projects.
```
Comment (jordynmackool) [inline proposals/NNNN-cloud_client_library_phase_1.md]: Change the word `device` to `define`

**Comment by joeygrover** [comment]
@jordynmackool I have updated the proposal per your review. Please review as time permits.

**Pull request merged**